### ISSUE TITLE
NullPointerException in io.micrometer.common.KeyValues

### ISSUE THREAD

**Issue by dietzsch** [micrometer-metrics/micrometer#3849]
Hello,

we have a set of Java Spring Boot applications, that we updated today to version 3.0.7. For monitoring the applications we have added dependency "io.micrometer:micrometer-registry-prometheus" to get the additional prometheus actuator. For all the Spring Boot apps that are running on the Webflux stack - use Netty - we see from time to time the following error in the logs:

```
2023-05-19 13:03:32 ERROR org.springframework.web.server.adapter.HttpWebHandlerAdapter [2cbc5a29-377] Error [java.lang.NullPointerException: Cannot invoke "java.lang.Comparable.compareTo(Object)" because "[]" is null] for HTTP GET "/actuator/prometheus", but ServerHttpResponse already committed (200 OK)
2023-05-19 13:03:32 ERROR reactor.core.publisher.Operators Operator called default onErrorDropped
java.lang.NullPointerException: Cannot invoke "java.lang.Comparable.compareTo(Object)" because "[]" is null
at java.base/java.util.ComparableTimSort.countRunAndMakeAscending(Unknown Source)
Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException:
Error has been observed at the following site(s):
*__checkpoint ⇢ AuthorizationWebFilter [DefaultWebFilterChain]
*__checkpoint ⇢ ExceptionTranslationWebFilter [DefaultWebFilterChain]
*__checkpoint ⇢ LogoutWebFilter [DefaultWebFilterChain]
*__checkpoint ⇢ ServerRequestCacheWebFilter [DefaultWebFilterChain]
*__checkpoint ⇢ SecurityContextServerWebExchangeWebFilter [DefaultWebFilterChain]
*__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
*__checkpoint ⇢ ReactorContextWebFilter [DefaultWebFilterChain]
*__checkpoint ⇢ HttpHeaderWriterWebFilter [DefaultWebFilterChain]
Original Stack Trace:
at java.base/java.util.ComparableTimSort.countRunAndMakeAscending(Unknown Source)
at java.base/java.util.ComparableTimSort.sort(Unknown Source)
at java.base/java.util.Arrays.sort(Unknown Source)
at io.micrometer.common.KeyValues.(KeyValues.java:47)
at io.micrometer.common.KeyValues.of(KeyValues.java:267)
at io.micrometer.observation.Observation$Context.getLowCardinalityKeyValues(Observation.java:1094)
at io.micrometer.core.instrument.observation.DefaultMeterObservationHandler.createTags(DefaultMeterObservationHandler.java:87)
at io.micrometer.core.instrument.observation.DefaultMeterObservationHandler.onStart(DefaultMeterObservationHandler.java:52)
at io.micrometer.observation.ObservationHandler$FirstMatchingCompositeObservationHandler.onStart(ObservationHandler.java:149)
at io.micrometer.observation.SimpleObservation.notifyOnObservationStarted(SimpleObservation.java:232)
at io.micrometer.observation.SimpleObservation.start(SimpleObservation.java:167)
at org.springframework.security.web.server.ObservationWebFilterChainDecorator$AroundWebFilterObservation$SimpleAroundWebFilterObservation$ObservationReference.start(ObservationWebFilterChainDecorator.java:395)
at org.springframework.security.web.server.ObservationWebFilterChainDecorator$AroundWebFilterObservation$SimpleAroundWebFilterObservation.start(ObservationWebFilterChainDecorator.java:276)
at org.springframework.security.web.server.ObservationWebFilterChainDecorator$AroundWebFilterObservation$SimpleAroundWebFilterObservation.lambda$wrap$0(ObservationWebFilterChainDecorator.java:339)
```

**Comment by jonatan-ivanov** [comment]
Thank you for the report. Could you please provide a minimal sample java project to reproduce this issue so we can more easily investigate and ensure any fix is working properly for your use case?

I was able to reproduce two issues through `KeyValues` that might be related:
- https://github.com/micrometer-metrics/micrometer/issues/3851
- https://github.com/micrometer-metrics/micrometer/issues/3852

But I was not able to reproduce it through the Observation API (your use-case). It seems Spring Security somehow manages to add a `null` `KeyValue` so there is definitely a bug on Spring Security's side (but I don't know how it manages to do this).
Could you please open a new issue for Spring Security too?

**Comment by dietzsch** [comment]
Thank you for the quick response. I will strip up one of the apps to provide a sample, in parallel I will also create an issue with Spring Security.

Best regards,
Alex

**Comment by dietzsch** [comment]
I have created a stripped up [sample project](https://github.com/dietzsch/vmaas-dns).

**Comment by jonatan-ivanov** [comment]
This project is not really minimal and if I strip it down further, I'm not able to reproduce the issue, could you please give us a [minimal sample java project](https://stackoverflow.com/help/minimal-reproducible-example)?
(I doubt that checkstyle, pmd, owasp-dependency-ckeck, license-report, a non existent subproject, logic in build.gradle idea, eclipse, jacoco, extra repositories, lombok and groovy are all needed.)

Also, the details that were in the template when you opened the issue (environment, repro steps) would be useful too, could you please include those as well to ensure I'm using the same java version (and vendor) as well as I follow the same repro steps (I checked the actuator endpoints including prometheus)?

**Comment by dietzsch** [comment]
I removed those plugins and also lombok.

Java: 17.0.7+7 (Bellsoft)

In our environment we see this exception from time to time in our logs, while Prometheus is calling every 20 seconds the /actuator/prometheus endpoint. The actuator is secured with basic auth and the credentials you can find in the application.yml.

**Comment by jonatan-ivanov** [comment]
I'm sorry but I'm not sure the situation is much better. :(
I put quite some effort to clean things up in the project you provided and it seems after the changes I still need to clean a bunch of things up (see my long list above the unnecessary things) and I assume I would get the same result. I can push the cleanup to your repo and you can check if it works or not but I find this a little counter-productive.

Also, could you please take a look at the issue template? It asks for a bit more extra details than the java version. Btw my Java version is quite similar (Liberica/BellSoft 17.0.6), once the reproducer is minimal, I'm going to retry it with the exact same version.

This is what I did after the cleanup:
1. `gradle bootRun`
2. Called `actuator/prometheus` (used the credentials that got me in) and get a valid response
3. Checked the metrics endpoint too, it looked good

You wrote that you see the exception in prod from time to time, could you please help me with two things?
1. The project you provided is able to reproduce the issue in your local dev environment, right?
2. How frequent "time to time" is? I only tried sending in a few requests (<20), how much should I need to send in? ~5? ~100? 100+k?

**Comment by dietzsch** [comment]
Sorry for the delay. Regrading the sample, I'm not sure what you have in mind to further reduce. The sample contains only three classes. I assume you like to get rid of the Spring Boot and maybe replace by Spring Framework directly, but as this is in the context of Spring Security and this is configured via the auto configurer it would take me some time and not sure if that helps finally.
Regarding your question on the frequency, the applications (4 Spring Boot applications) are running 24x7 where the Prometheus is scraping the data every 20 seconds and in the logs I have seen this error 7 times within the last 4 days, so 100+k.
What I can say from my observations that I guess that the spring-boot-starter-webflux and the new netty stack is involved. We have further Spring Boot applications, which are using spring-boot-starter-web and use the same setup with micrometer and Prometheus, and there we have not seen any exception.
regarding setup, we have all the applications deployed on a Cloud Foundry running in our own data center. The Cloud Foundry is deployed on an Openstack infrastructure. We use the Java buildpack (https://github.com/cloudfoundry/java-buildpack/releases/tag/v4.58) from Cloud Foundry (version 4.58) with the mentioned Java version. The Prometheus is also running on Openstack, but outside of Cloud Foundry.
If I can provide any other information, please let me know.

Alex

PS: On the issue I opened for Spring Security there was the request to test with a newer Spring Security 6.0.4 snapshot version. So, seems they have already changed/fixed something in their code.

**Comment by jonatan-ivanov** [comment]
>I'm not sure what you have in mind to further reduce. The sample contains only three classes.

It's not about the number of classes (though I think `MicrometerJvmExtrasConfig` is not necessary) and it's more about the project config, I tried to explain it above (although they should not matter, the issue should not happen either but it does):
>(I doubt that checkstyle, pmd, owasp-dependency-ckeck, license-report, a non existent subproject, logic in build.gradle idea, eclipse, jacoco, extra repositories, lombok and groovy are all needed.)

>I assume you like to get rid of the Spring Boot and maybe replace by Spring Framework directly

No, not at all, Boot is completely fine (see what I meant above).

>Regarding your question on the frequency, the applications (4 Spring Boot applications) are running 24x7 where the Prometheus is scraping the data every 20 seconds and in the logs I have seen this error 7 times within the last 4 days, so 100+k.

Oh, I definitely did not send that many (though isn't it closer to (24*60*4)*(5*4)/7~=16+k?). That might explain why I was not able to reproduce it. With this application, were you able to repro the issue in your local dev environment? I assumed yes and since I was not able to, I got confused.

>What I can say from my observations that I guess that the spring-boot-starter-webflux and the new netty stack is involved. We have further Spring Boot applications, which are using spring-boot-starter-web and use the same setup with micrometer and Prometheus, and there we have not seen any exception.

This makes sense, the instrumentation is different for web and webflux so you can run into issues that is specific to one stack (webflux is much harder).

>PS: On the issue I opened for Spring Security there was the request to test with a newer Spring Security 6.0.4 snapshot version. So, seems they have already changed/fixed something in their code.

Thank you, I subscribed to the issue, what Josh saying about the concurrency issue makes sense to me, if upgrading fixes it, I will talk to him to find out how is it possible to cause such NPE on Micrometer side (we need to fix that too).

**Comment by jonatan-ivanov** [comment]
I'm closing this since it is superseded by
- https://github.com/micrometer-metrics/micrometer/issues/3851
- https://github.com/micrometer-metrics/micrometer/issues/3852
- https://github.com/spring-projects/spring-security/issues/13200

**Comment by maheshtallada** [comment]
Hello Everyone!

Is there any solution for the issue in discussion?

I'm facing a similar issue. Can anyone let me know the probable fix.


2024-01-04 16:48:45,895[${sys:domain},/] 2024-01-04 16:48:45,895[]  ERROR [scheduling-1] support.TaskUtils$LoggingErrorHandler (TaskUtils.java:95) - Unexpected error occurred in scheduled task
java.lang.NullPointerException: null
	at java.base/java.util.Objects.requireNonNull(Objects.java:209) ~[?:?]
	at io.micrometer.common.ImmutableKeyValue.<init>(ImmutableKeyValue.java:38) ~[micrometer-commons-1.11.7.jar:1.11.7]
	at io.micrometer.common.KeyValue.of(KeyValue.java:48) ~[micrometer-commons-1.11.7.jar:1.11.7]
	at io.micrometer.common.KeyValue.of(KeyValue.java:58) ~[micrometer-commons-1.11.7.jar:1.11.7]....

**Comment by dietzsch** [comment]
Hi @maheshtallada,

we are using 3.1.7 right now and with that version we don't see the NPE any longer. Actually there were some fixes in Spring Security that have resolved the issue.

**Comment by maheshtallada** [comment]
Hi @dietzsch,

Thanks for the info. However in my case, it doesn't seem to occur because of Spring Security. 
Also, providing the complete error as the error is same but occurring through a different module of micrometer. PFB the dependency tree and kindly let me know if any clue.

2024-01-04 18:09:12,002[${sys:domain},/] 2024-01-04 18:09:12,002[]  ERROR [scheduling-1] support.TaskUtils$LoggingErrorHandler (TaskUtils.java:95) - **Unexpected error occurred in scheduled task
java.lang.NullPointerException: null**
	at java.base/java.util.Objects.requireNonNull(Objects.java:209) ~[?:?]
	at io.micrometer.common.ImmutableKeyValue.<init>(ImmutableKeyValue.java:38) ~[micrometer-commons-1.12.0.jar:1.12.0]
	at io.micrometer.common.KeyValue.of(KeyValue.java:48) ~[micrometer-commons-1.12.0.jar:1.12.0]
	at io.micrometer.common.KeyValue.of(KeyValue.java:58) ~[micrometer-commons-1.12.0.jar:1.12.0]
	at **org.springframework.scheduling.support.DefaultScheduledTaskObservationConvention.codeNamespace(DefaultScheduledTaskObservationConvention.java:64) ~[spring-context-6.1.1.jar:6.1.1]**
	at org.springframework.scheduling.support.DefaultScheduledTaskObservationConvention.getLowCardinalityKeyValues(DefaultScheduledTaskObservationConvention.java:56) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.scheduling.support.DefaultScheduledTaskObservationConvention.getLowCardinalityKeyValues(DefaultScheduledTaskObservationConvention.java:31) ~[spring-context-6.1.1.jar:6.1.1]
	at io.micrometer.observation.SimpleObservation.start(SimpleObservation.java:152) ~[micrometer-observation-1.12.0.jar:1.12.0]
	at io.micrometer.observation.Observation.observe(Observation.java:497) ~[micrometer-observation-1.12.0.jar:1.12.0]
	at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:124) ~[spring-context-6.1.1.jar:6.1.1]
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) [spring-context-6.1.1.jar:6.1.1]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) [?:?]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.base/java.lang.Thread.run(Thread.java:833) [?:?]


Here's the dependency tree.

[[1;34mINFO[m] com.<org>.chub:sms-integration-api:jar:0.0.1-SNAPSHOT
[[1;34mINFO[m] +- com.<org>.chub:commhub-commons:jar:10.0.0:compile
[[1;34mINFO[m] |  +- com.<org>.chub:commons-utils:jar:4.0.0:compile
[[1;34mINFO[m] |  |  +- com.<org>.chub:commons-cloudconfig:jar:4.0.0:compile
[[1;34mINFO[m] |  |  +- org.springframework.integration:spring-integration-core:jar:6.1.4:compile
[[1;34mINFO[m] |  |  |  +- org.springframework:spring-messaging:jar:6.0.13:compile
[[1;34mINFO[m] |  |  |  \- org.springframework.retry:spring-retry:jar:2.0.4:compile
[[1;34mINFO[m] |  |  +- org.jasypt:jasypt:jar:1.9.3:compile
[[1;34mINFO[m] |  |  +- org.apache.directory.studio:org.apache.commons.codec:jar:1.8:compile
[[1;34mINFO[m] |  |  +- com.azure:azure-spring-data-cosmos:jar:5.7.0:compile
[[1;34mINFO[m] |  |  |  +- com.azure:azure-cosmos:jar:4.52.0:compile
[[1;34mINFO[m] |  |  |  |  +- com.fasterxml.jackson.module:jackson-module-afterburner:jar:2.15.3:compile
[[1;34mINFO[m] |  |  |  |  \- org.hdrhistogram:HdrHistogram:jar:2.1.12:compile
[[1;34mINFO[m] |  |  |  +- org.javatuples:javatuples:jar:1.2:compile
[[1;34mINFO[m] |  |  |  \- javax.annotation:javax.annotation-api:jar:1.3.2:compile
[[1;34mINFO[m] |  |  +- com.azure.spring:spring-cloud-azure-starter-keyvault-secrets:jar:5.7.0:compile
[[1;34mINFO[m] |  |  |  +- com.azure.spring:spring-cloud-azure-starter:jar:5.7.0:compile
[[1;34mINFO[m] |  |  |  |  \- com.azure.spring:spring-cloud-azure-autoconfigure:jar:5.7.0:compile
[[1;34mINFO[m] |  |  |  |     \- com.azure.spring:spring-cloud-azure-service:jar:5.7.0:compile
[[1;34mINFO[m] |  |  |  |        \- com.azure.spring:spring-cloud-azure-core:jar:5.7.0:compile
[[1;34mINFO[m] |  |  |  |           +- com.azure:azure-identity:jar:1.10.4:compile
[[1;34mINFO[m] |  |  |  |           |  +- com.microsoft.azure:msal4j:jar:1.13.9:compile
[[1;34mINFO[m] |  |  |  |           |  +- com.microsoft.azure:msal4j-persistence-extension:jar:1.2.0:compile
[[1;34mINFO[m] |  |  |  |           |  |  \- net.java.dev.jna:jna:jar:5.13.0:compile
[[1;34mINFO[m] |  |  |  |           |  \- net.java.dev.jna:jna-platform:jar:5.6.0:compile
[[1;34mINFO[m] |  |  |  |           +- com.azure:azure-core-amqp:jar:2.8.11:compile
[[1;34mINFO[m] |  |  |  |           \- com.azure:azure-core-management:jar:1.11.7:compile
[[1;34mINFO[m] |  |  |  \- com.azure:azure-security-keyvault-secrets:jar:4.7.1:compile
[[1;34mINFO[m] |  |  +- com.google.guava:guava:jar:32.1.3-jre:compile
[[1;34mINFO[m] |  |  |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[[1;34mINFO[m] |  |  |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[[1;34mINFO[m] |  |  |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[[1;34mINFO[m] |  |  |  +- org.checkerframework:checker-qual:jar:3.37.0:compile
[[1;34mINFO[m] |  |  |  +- com.google.errorprone:error_prone_annotations:jar:2.21.1:compile
[[1;34mINFO[m] |  |  |  \- com.google.j2objc:j2objc-annotations:jar:2.8:compile
[[1;34mINFO[m] |  |  +- org.springframework.cloud:spring-cloud-starter-config:jar:4.1.0:compile
[[1;34mINFO[m] |  |  |  \- org.springframework.cloud:spring-cloud-config-client:jar:4.1.0:compile
[[1;34mINFO[m] |  |  |     \- org.apache.httpcomponents.client5:httpclient5:jar:5.2.1:compile
[[1;34mINFO[m] |  |  |        +- org.apache.httpcomponents.core5:httpcore5:jar:5.2.3:compile
[[1;34mINFO[m] |  |  |        \- org.apache.httpcomponents.core5:httpcore5-h2:jar:5.2.3:compile
[[1;34mINFO[m] |  |  \- org.springframework.boot:spring-boot-autoconfigure:jar:3.1.5:compile
[[1;34mINFO[m] |  +- com.<org>.chub:commons-exceptions:jar:4.0.0:compile
[[1;34mINFO[m] |  |  +- org.springframework.boot:spring-boot-devtools:jar:3.1.5:runtime
[[1;34mINFO[m] |  |  +- org.apache.commons:commons-lang3:jar:3.12.0:compile
[[1;34mINFO[m] |  |  \- org.springframework.boot:spring-boot-starter-actuator:jar:3.1.5:compile
[[1;34mINFO[m] |  |     +- org.springframework.boot:spring-boot-actuator-autoconfigure:jar:3.1.5:compile
[[1;34mINFO[m] |  |     |  \- org.springframework.boot:spring-boot-actuator:jar:3.1.5:compile
[[1;34mINFO[m] |  |     \- io.micrometer:micrometer-core:jar:1.11.5:compile
[[1;34mINFO[m] |  |        \- org.latencyutils:LatencyUtils:jar:2.0.3:runtime
[[1;34mINFO[m] |  +- com.<org>.chub:commons-stubs:jar:4.0.0:compile
[[1;34mINFO[m] |  |  \- org.apache.commons:commons-collections4:jar:4.4:compile
[[1;34mINFO[m] |  +- org.apache.cxf:cxf-rt-frontend-jaxws:jar:4.0.3:compile
[[1;34mINFO[m] |  |  +- xml-resolver:xml-resolver:jar:1.2:compile
[[1;34mINFO[m] |  |  +- org.ow2.asm:asm:jar:9.5:compile
[[1;34mINFO[m] |  |  +- org.apache.cxf:cxf-rt-bindings-soap:jar:4.0.3:compile
[[1;34mINFO[m] |  |  |  +- jakarta.xml.soap:jakarta.xml.soap-api:jar:3.0.0:compile
[[1;34mINFO[m] |  |  |  +- jakarta.jws:jakarta.jws-api:jar:3.0.0:compile
[[1;34mINFO[m] |  |  |  +- jakarta.xml.ws:jakarta.xml.ws-api:jar:4.0.0:compile
[[1;34mINFO[m] |  |  |  +- org.apache.cxf:cxf-rt-wsdl:jar:4.0.3:compile
[[1;34mINFO[m] |  |  |  |  \- wsdl4j:wsdl4j:jar:1.6.3:compile
[[1;34mINFO[m] |  |  |  \- org.apache.cxf:cxf-rt-databinding-jaxb:jar:4.0.3:compile
[[1;34mINFO[m] |  |  +- org.apache.cxf:cxf-rt-bindings-xml:jar:4.0.3:compile
[[1;34mINFO[m] |  |  +- org.apache.cxf:cxf-rt-frontend-simple:jar:4.0.3:compile
[[1;34mINFO[m] |  |  |  \- org.eclipse.angus:angus-mail:jar:1.1.0:compile
[[1;34mINFO[m] |  |  |     \- jakarta.mail:jakarta.mail-api:jar:2.1.2:compile
[[1;34mINFO[m] |  |  +- org.apache.cxf:cxf-rt-ws-addr:jar:4.0.3:compile
[[1;34mINFO[m] |  |  |  \- org.apache.cxf:cxf-rt-ws-policy:jar:4.0.3:compile
[[1;34mINFO[m] |  |  \- org.eclipse.angus:angus-activation:jar:2.0.1:compile
[[1;34mINFO[m] |  +- org.apache.cxf:cxf-rt-transports-http:jar:4.0.3:compile
[[1;34mINFO[m] |  +- org.apache.cxf:cxf-rt-ws-security:jar:4.0.3:compile
[[1;34mINFO[m] |  |  +- org.apache.cxf:cxf-rt-security-saml:jar:4.0.3:compile
[[1;34mINFO[m] |  |  |  \- org.apache.cxf:cxf-rt-security:jar:4.0.3:compile
[[1;34mINFO[m] |  |  +- org.ehcache:ehcache:jar:jakarta:3.10.8:compile
[[1;34mINFO[m] |  |  |  \- javax.cache:cache-api:jar:1.1.1:compile
[[1;34mINFO[m] |  |  +- org.apache.wss4j:wss4j-ws-security-dom:jar:3.0.1:compile
[[1;34mINFO[m] |  |  |  \- org.apache.wss4j:wss4j-ws-security-common:jar:3.0.1:compile
[[1;34mINFO[m] |  |  |     +- org.apache.santuario:xmlsec:jar:3.0.2:compile
[[1;34mINFO[m] |  |  |     +- org.opensaml:opensaml-saml-impl:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     |  +- org.opensaml:opensaml-core:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     |  +- org.opensaml:opensaml-profile-api:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     |  +- org.opensaml:opensaml-saml-api:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     |  +- org.opensaml:opensaml-security-api:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     |  +- org.opensaml:opensaml-security-impl:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     |  +- org.opensaml:opensaml-soap-api:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     |  +- org.opensaml:opensaml-xmlsec-api:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     |  +- org.opensaml:opensaml-xmlsec-impl:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     |  \- net.shibboleth.utilities:java-support:jar:8.4.0:compile
[[1;34mINFO[m] |  |  |     +- org.cryptacular:cryptacular:jar:1.2.5:compile
[[1;34mINFO[m] |  |  |     |  \- org.bouncycastle:bcprov-jdk18on:jar:1.71:compile
[[1;34mINFO[m] |  |  |     +- org.opensaml:opensaml-xacml-impl:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     |  \- org.opensaml:opensaml-xacml-api:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |     \- org.opensaml:opensaml-xacml-saml-impl:jar:4.3.0:compile
[[1;34mINFO[m] |  |  |        \- org.opensaml:opensaml-xacml-saml-api:jar:4.3.0:compile
[[1;34mINFO[m] |  |  +- org.apache.wss4j:wss4j-policy:jar:3.0.1:compile
[[1;34mINFO[m] |  |  |  \- org.apache.neethi:neethi:jar:3.2.0:compile
[[1;34mINFO[m] |  |  +- org.apache.wss4j:wss4j-ws-security-stax:jar:3.0.1:compile
[[1;34mINFO[m] |  |  |  \- org.apache.wss4j:wss4j-bindings:jar:3.0.1:compile
[[1;34mINFO[m] |  |  +- com.sun.xml.messaging.saaj:saaj-impl:jar:3.0.2:compile
[[1;34mINFO[m] |  |  |  \- org.jvnet.staxex:stax-ex:jar:2.1.0:compile
[[1;34mINFO[m] |  |  \- org.apache.wss4j:wss4j-ws-security-policy-stax:jar:3.0.1:compile
[[1;34mINFO[m] |  +- com.azure:azure-storage-blob:jar:12.25.0:compile
[[1;34mINFO[m] |  |  +- com.azure:azure-core:jar:1.45.0:compile
[[1;34mINFO[m] |  |  |  \- com.azure:azure-json:jar:1.1.0:compile
[[1;34mINFO[m] |  |  +- com.azure:azure-core-http-netty:jar:1.13.10:compile
[[1;34mINFO[m] |  |  |  +- io.netty:netty-transport-native-unix-common:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  |  +- io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.100.Final:compile
[[1;34mINFO[m] |  |  |  |  \- io.netty:netty-transport-classes-kqueue:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  |  \- io.netty:netty-tcnative-boringssl-static:jar:2.0.61.Final:compile
[[1;34mINFO[m] |  |  |     +- io.netty:netty-tcnative-classes:jar:2.0.61.Final:compile
[[1;34mINFO[m] |  |  |     +- io.netty:netty-tcnative-boringssl-static:jar:linux-x86_64:2.0.61.Final:compile
[[1;34mINFO[m] |  |  |     +- io.netty:netty-tcnative-boringssl-static:jar:linux-aarch_64:2.0.61.Final:compile
[[1;34mINFO[m] |  |  |     +- io.netty:netty-tcnative-boringssl-static:jar:osx-x86_64:2.0.61.Final:compile
[[1;34mINFO[m] |  |  |     +- io.netty:netty-tcnative-boringssl-static:jar:osx-aarch_64:2.0.61.Final:compile
[[1;34mINFO[m] |  |  |     \- io.netty:netty-tcnative-boringssl-static:jar:windows-x86_64:2.0.61.Final:compile
[[1;34mINFO[m] |  |  +- com.azure:azure-storage-common:jar:12.24.0:compile
[[1;34mINFO[m] |  |  +- com.azure:azure-storage-internal-avro:jar:12.10.0:compile
[[1;34mINFO[m] |  |  \- com.fasterxml.jackson.dataformat:jackson-dataformat-xml:jar:2.15.3:compile
[[1;34mINFO[m] |  +- org.projectlombok:lombok:jar:1.18.30:compile
[[1;34mINFO[m] |  +- org.springdoc:springdoc-openapi-ui:jar:1.7.0:compile
[[1;34mINFO[m] |  |  \- org.webjars:swagger-ui:jar:4.18.2:compile
[[1;34mINFO[m] |  +- org.springdoc:springdoc-openapi-webmvc-core:jar:1.7.0:compile
[[1;34mINFO[m] |  |  \- org.springdoc:springdoc-openapi-common:jar:1.7.0:compile
[[1;34mINFO[m] |  |     \- io.swagger.core.v3:swagger-core:jar:2.2.9:compile
[[1;34mINFO[m] |  |        +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.15.3:compile
[[1;34mINFO[m] |  |        +- io.swagger.core.v3:swagger-models:jar:2.2.9:compile
[[1;34mINFO[m] |  |        \- jakarta.validation:jakarta.validation-api:jar:3.0.2:compile
[[1;34mINFO[m] |  +- org.springframework.boot:spring-boot-starter-data-redis:jar:3.1.5:compile
[[1;34mINFO[m] |  |  +- org.springframework.data:spring-data-redis:jar:3.1.5:compile
[[1;34mINFO[m] |  |  |  +- org.springframework.data:spring-data-keyvalue:jar:3.1.5:compile
[[1;34mINFO[m] |  |  |  +- org.springframework:spring-oxm:jar:6.0.13:compile
[[1;34mINFO[m] |  |  |  \- org.springframework:spring-context-support:jar:6.0.13:compile
[[1;34mINFO[m] |  |  \- io.lettuce:lettuce-core:jar:6.2.6.RELEASE:compile
[[1;34mINFO[m] |  +- com.squareup.retrofit2:retrofit:jar:2.9.0:compile
[[1;34mINFO[m] |  |  \- com.squareup.okhttp3:okhttp:jar:4.10.0:compile
[[1;34mINFO[m] |  |     +- com.squareup.okio:okio-jvm:jar:3.0.0:compile
[[1;34mINFO[m] |  |     |  +- org.jetbrains.kotlin:kotlin-stdlib-jdk8:jar:1.8.22:compile
[[1;34mINFO[m] |  |     |  |  \- org.jetbrains.kotlin:kotlin-stdlib-jdk7:jar:1.8.22:compile
[[1;34mINFO[m] |  |     |  \- org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.8.22:compile
[[1;34mINFO[m] |  |     \- org.jetbrains.kotlin:kotlin-stdlib:jar:1.8.22:compile
[[1;34mINFO[m] |  |        \- org.jetbrains:annotations:jar:13.0:compile
[[1;34mINFO[m] |  +- commons-io:commons-io:jar:2.15.0:compile
[[1;34mINFO[m] |  +- redis.clients:jedis:jar:4.3.2:compile
[[1;34mINFO[m] |  |  +- org.apache.commons:commons-pool2:jar:2.11.1:compile
[[1;34mINFO[m] |  |  +- org.json:json:jar:20220320:compile
[[1;34mINFO[m] |  |  \- com.google.code.gson:gson:jar:2.10.1:compile
[[1;34mINFO[m] |  +- org.springframework.cloud:spring-cloud-starter-bootstrap:jar:4.0.4:compile
[[1;34mINFO[m] |  |  \- org.springframework.cloud:spring-cloud-starter:jar:4.0.4:compile
[[1;34mINFO[m] |  |     +- org.springframework.cloud:spring-cloud-context:jar:4.0.4:compile
[[1;34mINFO[m] |  |     |  \- org.springframework.security:spring-security-crypto:jar:6.1.5:compile
[[1;34mINFO[m] |  |     +- org.springframework.cloud:spring-cloud-commons:jar:4.0.4:compile
[[1;34mINFO[m] |  |     \- org.springframework.security:spring-security-rsa:jar:1.0.12.RELEASE:compile
[[1;34mINFO[m] |  |        \- org.bouncycastle:bcpkix-jdk18on:jar:1.73:compile
[[1;34mINFO[m] |  |           \- org.bouncycastle:bcutil-jdk18on:jar:1.73:compile
[[1;34mINFO[m] |  +- com.microsoft.azure:azure-cosmosdb:jar:2.6.16:compile
[[1;34mINFO[m] |  |  +- com.microsoft.azure:azure-cosmosdb-commons:jar:2.6.16:compile
[[1;34mINFO[m] |  |  +- com.microsoft.azure:azure-cosmosdb-gateway:jar:2.6.16:compile
[[1;34mINFO[m] |  |  +- com.microsoft.azure:azure-cosmosdb-direct:jar:2.6.16:compile
[[1;34mINFO[m] |  |  |  \- io.dropwizard.metrics:metrics-core:jar:4.2.21:compile
[[1;34mINFO[m] |  |  +- com.fasterxml.uuid:java-uuid-generator:jar:3.1.4:compile
[[1;34mINFO[m] |  |  +- com.github.davidmoten:rxjava-extras:jar:0.8.0.17:compile
[[1;34mINFO[m] |  |  +- io.reactivex:rxjava:jar:1.3.8:compile
[[1;34mINFO[m] |  |  +- io.reactivex:rxjava-string:jar:1.1.1:compile
[[1;34mINFO[m] |  |  +- io.reactivex:rxnetty:jar:0.4.20:compile
[[1;34mINFO[m] |  |  +- io.netty:netty-handler:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  |  \- io.netty:netty-resolver:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  \- io.netty:netty-transport:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  +- org.apache.commons:commons-text:jar:1.11.0:compile
[[1;34mINFO[m] |  +- org.apache.cxf:cxf-core:jar:4.0.3:compile
[[1;34mINFO[m] |  |  +- jakarta.annotation:jakarta.annotation-api:jar:2.1.1:compile
[[1;34mINFO[m] |  |  +- org.glassfish.jaxb:jaxb-runtime:jar:4.0.3:compile
[[1;34mINFO[m] |  |  |  \- org.glassfish.jaxb:jaxb-core:jar:4.0.3:compile
[[1;34mINFO[m] |  |  |     +- org.glassfish.jaxb:txw2:jar:4.0.3:compile
[[1;34mINFO[m] |  |  |     \- com.sun.istack:istack-commons-runtime:jar:4.1.2:compile
[[1;34mINFO[m] |  |  \- org.apache.ws.xmlschema:xmlschema-core:jar:2.3.1:compile
[[1;34mINFO[m] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.15.3:compile
[[1;34mINFO[m] |  |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.15.3:compile
[[1;34mINFO[m] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.15.3:compile
[[1;34mINFO[m] |  +- com.fasterxml.woodstox:woodstox-core:jar:6.5.1:compile
[[1;34mINFO[m] |  |  \- org.codehaus.woodstox:stax2-api:jar:4.2.1:compile
[[1;34mINFO[m] |  +- com.google.protobuf:protobuf-java:jar:3.25.1:compile
[[1;34mINFO[m] |  +- io.netty:netty-codec-http:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  +- io.netty:netty-common:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  \- io.netty:netty-buffer:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  +- io.netty:netty-codec:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  +- org.springframework.data:spring-data-rest-core:jar:4.1.5:compile
[[1;34mINFO[m] |  |  +- org.springframework:spring-tx:jar:6.0.13:compile
[[1;34mINFO[m] |  |  +- org.springframework.hateoas:spring-hateoas:jar:2.1.2:compile
[[1;34mINFO[m] |  |  +- org.springframework.data:spring-data-commons:jar:3.1.5:compile
[[1;34mINFO[m] |  |  +- org.springframework.plugin:spring-plugin-core:jar:3.0.0:compile
[[1;34mINFO[m] |  |  +- org.atteo:evo-inflector:jar:1.3:compile
[[1;34mINFO[m] |  |  \- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.15.3:compile
[[1;34mINFO[m] |  +- io.projectreactor.netty:reactor-netty-http:jar:1.1.12:compile
[[1;34mINFO[m] |  |  +- io.netty:netty-codec-http2:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  +- io.netty:netty-resolver-dns:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  |  \- io.netty:netty-codec-dns:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  +- io.netty:netty-resolver-dns-native-macos:jar:osx-x86_64:4.1.100.Final:compile
[[1;34mINFO[m] |  |  |  \- io.netty:netty-resolver-dns-classes-macos:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  +- io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.100.Final:compile
[[1;34mINFO[m] |  |  |  \- io.netty:netty-transport-classes-epoll:jar:4.1.100.Final:compile
[[1;34mINFO[m] |  |  \- io.projectreactor:reactor-core:jar:3.5.11:compile
[[1;34mINFO[m] |  |     \- org.reactivestreams:reactive-streams:jar:1.0.4:compile
[[1;34mINFO[m] |  +- io.projectreactor.netty:reactor-netty:jar:1.1.12:compile
[[1;34mINFO[m] |  |  \- io.projectreactor.netty.incubator:reactor-netty-incubator-quic:jar:0.1.12:runtime
[[1;34mINFO[m] |  |     \- io.netty.incubator:netty-incubator-codec-native-quic:jar:linux-x86_64:0.0.51.Final:runtime
[[1;34mINFO[m] |  |        \- io.netty.incubator:netty-incubator-codec-classes-quic:jar:0.0.51.Final:runtime
[[1;34mINFO[m] |  \- io.projectreactor.netty:reactor-netty-core:jar:1.1.12:compile
[[1;34mINFO[m] |     \- io.netty:netty-handler-proxy:jar:4.1.100.Final:compile
[[1;34mINFO[m] |        \- io.netty:netty-codec-socks:jar:4.1.100.Final:compile
[[1;34mINFO[m] +- com.microsoft.azure:azure-eventhubs:jar:3.3.0:compile
[[1;34mINFO[m] |  +- org.apache.qpid:proton-j:jar:0.33.8:compile
[[1;34mINFO[m] |  +- com.microsoft.azure:qpid-proton-j-extensions:jar:1.2.4:compile
[[1;34mINFO[m] |  +- org.slf4j:slf4j-api:jar:2.0.9:compile
[[1;34mINFO[m] |  \- com.microsoft.azure:azure-client-authentication:jar:1.7.3:compile
[[1;34mINFO[m] |     +- com.microsoft.azure:azure-client-runtime:jar:1.7.3:compile
[[1;34mINFO[m] |     |  \- com.microsoft.rest:client-runtime:jar:1.7.3:compile
[[1;34mINFO[m] |     |     +- com.squareup.okhttp3:logging-interceptor:jar:4.10.0:compile
[[1;34mINFO[m] |     |     +- com.squareup.okhttp3:okhttp-urlconnection:jar:4.10.0:compile
[[1;34mINFO[m] |     |     +- com.squareup.retrofit2:converter-jackson:jar:2.7.2:compile
[[1;34mINFO[m] |     |     +- com.fasterxml.jackson.datatype:jackson-datatype-joda:jar:2.15.3:compile
[[1;34mINFO[m] |     |     |  \- joda-time:joda-time:jar:2.10.14:compile
[[1;34mINFO[m] |     |     \- com.squareup.retrofit2:adapter-rxjava:jar:2.7.2:compile
[[1;34mINFO[m] |     +- commons-codec:commons-codec:jar:1.15:compile
[[1;34mINFO[m] |     +- com.microsoft.azure:adal4j:jar:1.6.4:compile
[[1;34mINFO[m] |     |  \- com.nimbusds:oauth2-oidc-sdk:jar:6.5:compile
[[1;34mINFO[m] |     |     \- com.sun.mail:javax.mail:jar:1.6.1:compile
[[1;34mINFO[m] |     |        \- javax.activation:activation:jar:1.1:compile
[[1;34mINFO[m] |     \- com.microsoft.azure:azure-annotations:jar:1.10.0:compile
[[1;34mINFO[m] +- com.googlecode.json-simple:json-simple:jar:1.1.1:compile
[[1;34mINFO[m] +- org.springframework.boot:spring-boot-starter-web:jar:3.1.5:compile
[[1;34mINFO[m] |  +- org.springframework.boot:spring-boot-starter:jar:3.1.5:compile
[[1;34mINFO[m] |  |  +- org.springframework.boot:spring-boot:jar:3.1.5:compile
[[1;34mINFO[m] |  |  \- org.yaml:snakeyaml:jar:1.33:compile
[[1;34mINFO[m] |  +- org.springframework.boot:spring-boot-starter-json:jar:3.1.5:compile
[[1;34mINFO[m] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.15.3:compile
[[1;34mINFO[m] |  |  \- com.fasterxml.jackson.module:jackson-module-parameter-names:jar:2.15.3:compile
[[1;34mINFO[m] |  +- org.springframework.boot:spring-boot-starter-tomcat:jar:3.1.5:compile
[[1;34mINFO[m] |  |  +- org.apache.tomcat.embed:tomcat-embed-core:jar:10.1.15:compile
[[1;34mINFO[m] |  |  +- org.apache.tomcat.embed:tomcat-embed-el:jar:10.1.15:compile
[[1;34mINFO[m] |  |  \- org.apache.tomcat.embed:tomcat-embed-websocket:jar:10.1.15:compile
[[1;34mINFO[m] |  +- org.springframework:spring-web:jar:6.0.13:compile
[[1;34mINFO[m] |  |  +- org.springframework:spring-beans:jar:6.0.13:compile
[[1;34mINFO[m] |  |  \- io.micrometer:micrometer-observation:jar:1.11.5:compile
[[1;34mINFO[m] |  |     \- io.micrometer:micrometer-commons:jar:1.11.5:compile
[[1;34mINFO[m] |  \- org.springframework:spring-webmvc:jar:6.0.13:compile
[[1;34mINFO[m] |     +- org.springframework:spring-aop:jar:6.0.13:compile
[[1;34mINFO[m] |     +- org.springframework:spring-context:jar:6.0.13:compile
[[1;34mINFO[m] |     \- org.springframework:spring-expression:jar:6.0.13:compile
[[1;34mINFO[m] +- org.apache.logging.log4j:log4j-core:jar:2.20.0:compile
[[1;34mINFO[m] +- org.apache.logging.log4j:log4j-api:jar:2.20.0:compile
[[1;34mINFO[m] +- org.apache.logging.log4j:log4j-slf4j-impl:jar:2.20.0:compile
[[1;34mINFO[m] +- org.apache.logging.log4j:log4j-jul:jar:2.20.0:compile
[[1;34mINFO[m] +- org.springframework.boot:spring-boot-starter-test:jar:3.1.5:test
[[1;34mINFO[m] |  +- org.springframework.boot:spring-boot-test:jar:3.1.5:test
[[1;34mINFO[m] |  +- org.springframework.boot:spring-boot-test-autoconfigure:jar:3.1.5:test
[[1;34mINFO[m] |  +- com.jayway.jsonpath:json-path:jar:2.8.0:compile
[[1;34mINFO[m] |  +- jakarta.xml.bind:jakarta.xml.bind-api:jar:4.0.1:compile
[[1;34mINFO[m] |  |  \- jakarta.activation:jakarta.activation-api:jar:2.1.2:compile
[[1;34mINFO[m] |  +- org.assertj:assertj-core:jar:3.24.2:test
[[1;34mINFO[m] |  |  \- net.bytebuddy:byte-buddy:jar:1.14.9:compile
[[1;34mINFO[m] |  +- org.hamcrest:hamcrest:jar:2.2:test
[[1;34mINFO[m] |  +- org.junit.jupiter:junit-jupiter:jar:5.9.3:test
[[1;34mINFO[m] |  |  +- org.junit.jupiter:junit-jupiter-api:jar:5.9.3:test
[[1;34mINFO[m] |  |  |  +- org.opentest4j:opentest4j:jar:1.2.0:test
[[1;34mINFO[m] |  |  |  +- org.junit.platform:junit-platform-commons:jar:1.9.3:test
[[1;34mINFO[m] |  |  |  \- org.apiguardian:apiguardian-api:jar:1.1.2:test
[[1;34mINFO[m] |  |  +- org.junit.jupiter:junit-jupiter-params:jar:5.9.3:test
[[1;34mINFO[m] |  |  \- org.junit.jupiter:junit-jupiter-engine:jar:5.9.3:test
[[1;34mINFO[m] |  |     \- org.junit.platform:junit-platform-engine:jar:1.9.3:test
[[1;34mINFO[m] |  +- org.mockito:mockito-core:jar:5.3.1:test
[[1;34mINFO[m] |  |  +- net.bytebuddy:byte-buddy-agent:jar:1.14.9:test
[[1;34mINFO[m] |  |  \- org.objenesis:objenesis:jar:3.3:test
[[1;34mINFO[m] |  +- org.mockito:mockito-junit-jupiter:jar:5.3.1:test
[[1;34mINFO[m] |  +- org.skyscreamer:jsonassert:jar:1.5.1:test
[[1;34mINFO[m] |  |  \- com.vaadin.external.google:android-json:jar:0.0.20131108.vaadin1:test
[[1;34mINFO[m] |  +- org.springframework:spring-core:jar:6.0.13:compile
[[1;34mINFO[m] |  |  \- org.springframework:spring-jcl:jar:6.0.13:compile
[[1;34mINFO[m] |  +- org.springframework:spring-test:jar:6.0.13:test
[[1;34mINFO[m] |  \- org.xmlunit:xmlunit-core:jar:2.9.1:test
[[1;34mINFO[m] +- io.springfox:springfox-swagger-ui:jar:3.0.0:compile
[[1;34mINFO[m] |  \- io.springfox:springfox-spring-webmvc:jar:3.0.0:compile
[[1;34mINFO[m] |     \- io.springfox:springfox-core:jar:3.0.0:compile
[[1;34mINFO[m] +- io.springfox:springfox-swagger2:jar:3.0.0:compile
[[1;34mINFO[m] |  +- io.springfox:springfox-spi:jar:3.0.0:compile
[[1;34mINFO[m] |  +- io.springfox:springfox-schema:jar:3.0.0:compile
[[1;34mINFO[m] |  +- io.springfox:springfox-swagger-common:jar:3.0.0:compile
[[1;34mINFO[m] |  |  \- io.swagger.core.v3:swagger-annotations:jar:2.1.2:compile
[[1;34mINFO[m] |  +- io.springfox:springfox-spring-web:jar:3.0.0:compile
[[1;34mINFO[m] |  |  \- io.github.classgraph:classgraph:jar:4.8.83:compile
[[1;34mINFO[m] |  +- io.springfox:springfox-spring-webflux:jar:3.0.0:compile
[[1;34mINFO[m] |  +- com.fasterxml:classmate:jar:1.5.1:compile
[[1;34mINFO[m] |  +- org.springframework.plugin:spring-plugin-metadata:jar:2.0.0.RELEASE:compile
[[1;34mINFO[m] |  +- io.swagger:swagger-annotations:jar:1.5.20:compile
[[1;34mINFO[m] |  +- io.swagger:swagger-models:jar:1.5.20:compile
[[1;34mINFO[m] |  \- org.mapstruct:mapstruct:jar:1.3.1.Final:runtime
[[1;34mINFO[m] +- io.springfox:springfox-bean-validators:jar:3.0.0:compile
[[1;34mINFO[m] +- org.springframework.boot:spring-boot-starter-log4j2:jar:3.1.5:compile
[[1;34mINFO[m] |  \- org.apache.logging.log4j:log4j-slf4j2-impl:jar:2.20.0:compile
[[1;34mINFO[m] +- com.microsoft.azure:applicationinsights-spring-boot-starter:jar:2.6.4:compile
[[1;34mINFO[m] |  \- com.microsoft.azure:applicationinsights-web:jar:2.6.4:compile
[[1;34mINFO[m] +- junit:junit:jar:4.13.2:test
[[1;34mINFO[m] |  \- org.hamcrest:hamcrest-core:jar:2.2:test
[[1;34mINFO[m] +- com.microsoft.azure:applicationinsights-logging-log4j2:jar:2.6.4:compile
[[1;34mINFO[m] |  \- com.microsoft.azure:applicationinsights-core:jar:2.6.4:compile
[[1;34mINFO[m] +- com.nimbusds:lang-tag:jar:1.4.4:compile
[[1;34mINFO[m] +- com.nimbusds:nimbus-jose-jwt:jar:9.18:compile
[[1;34mINFO[m] |  \- com.github.stephenc.jcip:jcip-annotations:jar:1.0-1:compile
[[1;34mINFO[m] \- net.minidev:json-smart:jar:2.4.11:compile
[[1;34mINFO[m]    \- net.minidev:accessors-smart:jar:2.4.11:compile

**Comment by jonatan-ivanov** [comment]
@maheshtallada this might be an issue with the instrumentation of `@Scheduled` in Spring Framework. Could you please open an issue there?

**Comment by bclozel** [comment]
It might be already done in https://github.com/spring-projects/spring-framework/issues/31918
